internal/controller/dnsrecord_controller.go

```diff
@@ -81,6 +80,9 @@ func (r *DNSRecordReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		logger.Info("Removing Finalizer", "name", DNSRecordFinalizer)
 		controllerutil.RemoveFinalizer(dnsRecord, DNSRecordFinalizer)
 		if err = r.Update(ctx, dnsRecord); client.IgnoreNotFound(err) != nil {
+			if apierrors.IsConflict(err) {
```

Comment:
Do you think it's worth creating a generalised handler for this, in the DNS Operator, to help us reduce the number of calls to the API? Something like:
```
func PersistentUpdate(ctx, client, object, tries) error {

    err := client.Update(ctx, object)

    while err != nil && count < tries {
        count++

        // get fresh copy of object and generate patch

        err := client.Patch(ctx, patch)
    }
}
```

Reply:
Without some mechanism like that, every requeue could result in a few hits to the DNS Provider API, right?

Reply:
Possibly, but a bit out of scope for these particular changes. The change here is just preventing a conflict being logged as an error (the logs get littered with conflict errors that really are not that important) and doesn't cause any more or less API calls. We do this in a few places, but was missing here.

We did talk about some mechanism like you describe to limit API calls in particular cases, conflicts being one of the main ones, but it's probably it's own task as part of the [dnsrecord lifecycle](https://github.com/Kuadrant/dns-operator/labels/dnsrecord%20lifecycle) issues no?.